httpx/models.py

```diff
@@ -61,6 +61,7 @@
 AuthTypes = typing.Union[
     typing.Tuple[typing.Union[str, bytes], typing.Union[str, bytes]],
     typing.Callable[["AsyncRequest"], "AsyncRequest"],
+    "BaseMiddleware",  # type: ignore
```

Comment:
If we ignore this to prevent circular imports, consider adding this import:

```python
if typing.TYPE_CHECKING:
    from .middleware.base import BaseMiddleware
```

so that mypy can resolve "BaseMiddleware".

Reply:
Yes, that's why it's there. I did consider that option but seemed odd to add an `if` statement for it, but if that is The Right Thing™️  I'm happy to add it 🙂

Reply:
Hmm, that's annoying, adding that if upsets flake8 becuase `BaseMiddleware` is not being used 😕

Reply:
I think it’s actually a pretty standard pattern. I’ve been using it in various places where I’d only need to import something for typing purposes. Mypy recommends it here: https://mypy.readthedocs.io/en/latest/common_issues.html#import-cycles

Reply:
Doesn't look like it does, I've ignored that error on that line for now. Thanks 👍